packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SushiSuccessTokenLongShortPairFinancialProductLibrary.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+import "./LongShortPairFinancialProductLibrary.sol";
+import "../../../../common/implementation/Lockable.sol";
+
+/**
+ * @title Success Token Long Short Pair Financial Product Library.
+ * @notice Adds settlement logic to create success token LSPs. A success token pays out 50% of collateral as a
+ * floor, with the remaining 50% functioning like an embedded covered call.
+ * If the settlement is below the strike price then longs are worth 50% of collateral.
+ * If the settlement is above the strike then the payout is equal to:
+ * 0.5 + (0.5 * (expiryPrice - strikePrice) / expiryPrice)
+ * For example, consider a covered call option collateralized in SUSHI, with a strike a price of $20,
+ * and collateralPerPair of 2.
+ * - If the price is less than $20 then the each long is worth 0.5 collateralPerPair and each short is worth 0.5
+ * collateralPerPair.
+ * - If the price is more than $20 then each long is worth 0.5 collateralPerPair plus 0.5 times the fraction of
+ * collateralPerPair that was in the money, and each short is worth the remaining collateralPerPair.
+ * - Say settlement price is $30.  Then expiryPercentLong = 0.5 + (0.5 * (30 - 20) / 30) = 0.6667.
+ * If the collateralPerPair is 2, that means the long payout is 0.6667*2 = 1.3333 $SUSHI, which at a settlement
+ * price of $30 is worth $40. This is equivalent to the value of 1 $SUSHI plus the value of the $20 strike
+ * embedded call.
+ */
+contract SuccessTokenLongShortPairFinancialProductLibrary is LongShortPairFinancialProductLibrary, Lockable {
```

Comment:
File name should be updated to reflect the contract name

Reply:
good catch